### PR TITLE
add publication and details to dataset (backend only)

### PR DESCRIPTION
**Pull request by fm3**
### URL of deployed dev instance (used for testing):
- https://publications.webknossos.xyz

### Steps to test:
- add details and publication to a dataset in your database, e.g.
- `insert into webknossos.publications(_id, publicationDate, imageUrl, description) values('5c3c9ec895010095014759fd',NOW(),'https://webknossos.org/images/oxalis.svg','This is a wonderful dummy publication, it has authors, it has a link, it has a doi number, those could go here');`
- `update webknossos.datasets set _publication = '5c3c9ec895010095014759fd', details='{"species":"Drosophila melanogaster"}' where _id = '<A-DATASET-ID>' ;` (edit dataset id!)
- dataset list request should contain matching json values

### Issues:
- backend part for #3515 

------
- ~~[ ] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)~~
- [x] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable
- ~~[ ] Updated [documentation](../blob/master/docs) if applicable~~
- ~~[ ] Needs datastore update after deployment~~
- [x] Ready for review

### CODE REVIEW

**Comment by daniel-wer** [comment]
Can we add a description to the publication as well, e.g. from demo: 
> Drawitsch, F., Karimi, A., Boergens, K.M., and Helmstaedter, M. FluoEM, virtual labeling of axons in 3-dimensional electron microscopy data for long-range connectomics. Elife. 2018 Aug 14;7. pii: e38976. doi: 10.7554/eLife.38976.

Currently, this is usually the dataset description, but imo it is actually the publication description. We would get rid of all the duplicated dataset descriptions this way, offering more space for more important dataset-specific information, like species, brain region, acquisition method, etc.

**Comment by normanrz** [comment]
@daniel-wer I think that is the `details` field of publication. Maybe we should rename to `description` for consistency.

**Comment by fm3** [comment]
@daniel-wer absolutely. There are the fields `_id, publicationDate, imageUrl, title, details` for publications in this PR. the steps to test don’t include them all as most are optional. The idea was that this field should contain markdown-enabled text for just that purpose

**Comment by fm3** [comment]
sure, I can rename it to description

**Comment by daniel-wer** [comment]
Ah I see, perfect! :tada:

File: app/assets/javascripts/test/snapshots/public/test-bundle/test/backend-snapshot-tests/datasets.e2e.js.md [hunk 1]
```diff
@@ -540,6 +555,7 @@ Generated by [AVA](https://ava.li).
         logoUrl: '/images/mpi-logos.svg',
         name: 'rgb',
         owningOrganization: 'Organization_X',
+        publication: null,
```
Comment (jstriebel) [inline app/assets/javascripts/test/snapshots/public/test-bundle/test/backend-snapshot-tests/datasets.e2e.js.md]: Should we maybe add a details and a publication example to the snapshot tests?

File: conf/evolutions/036-add-lastTaskTypeId-to-user.sql [hunk 2]
```diff
@@ -1,4 +1,4 @@
--- https://github.com/scalableminds/webknossos/pull/TODO
+-- https://github.com/scalableminds/webknossos/pull/3466
```
Comment (jstriebel) [inline conf/evolutions/036-add-lastTaskTypeId-to-user.sql]: :heart:

File: app/assets/javascripts/test/snapshots/public/test-bundle/test/backend-snapshot-tests/datasets.e2e.js.md (hunk 1 above)
Comment (jstriebel) [reply app/assets/javascripts/test/snapshots/public/test-bundle/test/backend-snapshot-tests/datasets.e2e.js.md]: True :+1: Then nevermind :shipit:

**Pull request merged**